Tasks/PublishSymbols/PublishSymbols.ps1

```diff
@@ -6,42 +6,53 @@ try {
+            # Construct the semaphore message.
+            $utcNow = (Get-Date).ToUniversalTime()
+            $semaphoreMessage = "Unpublish: True, Machine: $env:ComputerName, BuildUri: $env:Build_BuildUri, BuildNumber: $env:Build_BuildNumber, RepositoryName: $env:Build_Repository_Name, RepositoryUri: $env:Build_Repository_Uri, Team Project: $env:System_TeamProject, CollectionUri: $env:System_TeamFoundationCollectionUri at $utcNow UTC"
+
+            # Delete the symbol store transaction.
+            [string]$SymbolsPath = Get-VstsInput -Name 'SymbolsPath' -Require
+            [string]$TransactionId = Get-VstsInput -Name 'TransactionId' -Require
+            Import-Module -Name $PSScriptRoot\PublishHelpers\PublishHelpers.psm1
+            Invoke-UnpublishSymbols -Share $SymbolsPath -TransactionId $TransactionId -MaximumWaitTime $SymbolsMaximumWaitTime -SemaphoreMessage $SemaphoreMessage
+            return
+        }
+        # Get the inputs.
+        [string]$SymbolsPath = Get-VstsInput -Name 'SymbolsPath'
+
+        if ([string]$SourceFolder = (Get-VstsInput -Name 'SourceFolder') -and
+            $SourceFolder -ne (Get-VstsTaskVariable -Name 'Build.SourcesDirectory' -Require)) {
+            Write-Warning (Get-VstsLocString -Key SourceFolderDeprecated0 -ArgumentList $SourceFolder)
+        }
+
+        [string]$SymbolsProduct = Get-VstsInput -Name 'SymbolsProduct' -Default (Get-VstsTaskVariable -Name 'Build.DefinitionName' -Require)
+        [string]$SymbolsVersion = Get-VstsInput -Name 'SymbolsVersion' -Default (Get-VstsTaskVariable -Name 'Build.BuildNumber' -Require)
+        [string]$SymbolsArtifactName = Get-VstsInput -Name 'SymbolsArtifactName'
+    }
+    [bool]$SkipIndexing = -not (Get-VstsInput -Name 'IndexSources' -AsBool)
     [bool]$TreatNotIndexedAsWarning = Get-VstsInput -Name 'TreatNotIndexedAsWarning' -AsBool
+    [string]$SymbolsFolder = Get-VstsInput -Name 'SymbolsFolder' -Default (Get-VstsTaskVariable -Name 'Build.SourcesDirectory' -Require)
 
-    # Get the PDB file paths.
-    $pdbFiles = @(Find-VstsFiles -LiteralDirectory $SymbolsFolder -LegacyPattern $SearchPattern)
-    Write-Host (Get-VstsLocString -Key Found0Files -ArgumentList $pdbFiles.Count)
+    if ( ($SymbolServerType -eq "FileShare") -or ($SymbolServerType -eq "TeamServices") -or (-not $SkipIndexing) ) {
+        # Get the PDB file paths.
+        [string]$SearchPattern = Get-VstsInput -Name 'SearchPattern' -Default "**\bin\**\*.pdb"
+        # $pdbFiles = @(Find-VstsFiles -LiteralDirectory $SymbolsFolder -LegacyPattern $SearchPattern)
+        $pdbFiles = @(Find-VstsMatch -DefaultRoot $SymbolsFolder -Pattern $SearchPattern)
+        Write-Host (Get-VstsLocString -Key Found0Files -ArgumentList $pdbFiles.Count)
+        
+        if (-not $pdbFiles -and $SearchPattern.Contains(';')) {
+            throw "No files found, You might want to use newlines instead of ';' to seperate search patterns"
```

Comment:
Slight tweaking: "No files found. Use newlines instead of ';' to separate search patterns."